src/farmer.rs

```diff
@@ -162,6 +170,12 @@ impl Builder {
                 allocated_space,
                 progress,
                 solutions,
+                initial_plotting_progress: Arc::new(Mutex::new(InitialPlottingProgress {
+                    starting_sector: single_disk_plot.plotted_sectors_count(),
```

Comment:
starting sector and current sector are the same?

Reply:
Only for the start. After start current sector will be increasing.

Reply:
what happens if I close the program in the midst of initial plotting (say at 50%), and then reopen it tomorrow? Will it start from 50% ?

Reply:
Line 320 -> I understand it will increment the `initial_plotting_progress` by 1 for the each notification.

However, if I close the program (say, after 50% progress), the value inside the mutex will be gone, it is not written to a file. And if I restart the program, `initial_plotting_progress` will be 0 again, although it should have been 50%.

Could you show me the code piece or the logic for how it will start from 50% ?

Reply:
If instead, `initial_plotting_progress` was overwritten by the value of the `current_sector` id or index for example, it would be more resilient to all scenarios I believe.

For example, I've done some plotting on my main computer to my external drive. Then I plugged my external drive to some another computer, and it would resume the initial plotting progress from where it's exactly left at.

But I can't see a way for this code to detect the prior progress.

Reply:
It works like this:
```rust
// Nothing is plotted. We need to plot 100 sectors
InitialPlottingProgress {
    starting_sector: 0,
    current_sector: 0,
    total_sectors: 100,
}
// Some sectors were plotted
InitialPlottingProgress {
    starting_sector: 0,
    current_sector: 10,
    total_sectors: 100,
}
// Farmer was stopped and started again
InitialPlottingProgress {
    starting_sector: 10,
    current_sector: 10,
    total_sectors: 100,
}
// Some sectors were plotted after that
InitialPlottingProgress {
    starting_sector: 10,
    current_sector: 20,
    total_sectors: 100,
}
```

Reply:
Ok, I missed the part where the monorepo returns the sector count via `plotted_sectors_count`. So, every time SDK farmer is initialized, it will update the `starting sector`.